code/datums/progressbar.dm

```diff
@@ -9,74 +9,208 @@
 	var/mob/user
 	var/client/client
 	var/listindex
+	var/image/shown_image
+	var/image/shown_image_darkened
+	var/alpha_filter
+	var/icon/alpha_icon
+	var/leftmost_pixel
+	var/rightmost_pixel
+	var/current_target
+	var/show_bar = TRUE
+	var/x_image_offset
+	var/y_image_offset
+	var/client/target_client
```

Comment:
Does this handle target client disconnect properly? (Set this to null, clear the progress bar and clean references)